airflow/jobs/backfill_job.py

```diff
@@ -218,6 +218,16 @@ def _update_counters(self, ti_status, session=None):
                 ti_status.running.pop(reduced_key)
                 ti_status.to_run[ti.key] = ti
 
+            elif ti.state == TaskInstanceState.SCHEDULED:
+                # Deferrable task can go from DEFERRED to SCHEDULED,
+                # when that happens, we need to remove it from running
+                # and add it to to_run
+                ti_status.running.pop(ti.key)
+                ti_status.to_run[ti.key] = ti
+
+            elif ti.state == TaskInstanceState.DEFERRED:
+                self.log.warning("The triggerer is not running. Please start the triggerer")
```

Comment:
Do we need to do some additional checks to ensure the triggerer is really not running? I’d imagine this could cause some data race issues where the triggerer _is_ running but just has some delays and does not pick up the task instance in time.

Reply:
Yeah. That's right

Reply:
What do you think about removing this check entirely? Looks like making a query to check if triggerer job is alive is too much?